cysync/src/store/database/helper/transaction/insert.ts

```diff
@@ -193,7 +196,8 @@ export const insertFromFullTxn = async (transaction: {
     await transactionDb.insert(newTxn);
   } else if (coin instanceof EthCoinData) {
     // Derive address from Xpub (It'll always give a mixed case address with checksum)
-    const myAddress = generateEthAddressFromXpub(xpub);
+    let myAddress = generateEthAddressFromXpub(xpub);
+    if (coin.coinListId === 0x0e) myAddress = formatHarmonyAddress(myAddress);
```

Comment:
Can we use the existing coin data instead of a hardcoded value? like `coin.coinListId === ETHCOINS['one']?.coinListId`